### ISSUE TITLE
$pnp.sp.search problem in IE11

### ISSUE THREAD

**Issue by joukamp** [SharePoint/PnP-JS-Core#354]
Thank you for reporting an issue, suggesting an enhancement, or asking a question. We appreciate your feedback - to help the team understand your
needs please complete the below template to ensure we have the details to help. Thanks!

#### Category
[ ] Enhancement

[ ] Bug

[x ] Question

#### Expected / Desired Behavior / Question
Hi
Envrionmet is Sharepoint 2013 on-premise and Sharepoint Online.

I have Content editor webpart with following:

`<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/fetch/2.0.2/fetch.min.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.min.js"></script>

<div id="searchDiv"></div>  

// for IE
 $pnp.setup({
 	headers: {
 		"Accept": "application/json;odata=verbose"
 	}
});

$pnp.sp.search('contenttype:newspage').then(function(result) {
	alert("SEARCH OK");
})
.catch(function(error){ 
	alert("ERROR");
});

#### Observed Behavior
I want "SEARCH OK" to happen. It works in Chrome but not in IE11, in ie its "ERROR"
Error is:  "**ReferenceError: 'Symbol' is undefined**"

But this next code does work in IE,Chrome 
$pnp.sp.profiles.myProperties.get().then(function(result) {
	alert("OKEI");
})
.catch(function(error){ 
	alert("ERROR");
});

What am I doing wrong here?
Thanks!
 Jouni

**Comment by tavikukko** [comment]
Have you included the polyfills?
Check this document: https://github.com/SharePoint/PnP-JS-Core/wiki/Getting-Started:-Install-&-Use

**Comment by tavikukko** [comment]
Oops, you had those already :)

**Comment by patrick-rodgers** [comment]
I think the issue might be with the library you are using as a fetch polyfill (https://github.com/github/fetch/blob/master/fetch.js) It looks like it is referencing Symbol and expecting it to be there. I am not sure why one works and not the other though. I checked and didn't see where we are using Symbol in the library. Can you share your whole code file (remove any PII details) so we can have a look?

**Comment by joukamp** [comment]
Here's my code, "Blank web part Page", "Seattle"-master:

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/fetch/2.0.2/fetch.min.js"></script>
<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.min.js"></script>

<div id="searchDiv"></div>   
<div id="user"></div>
  
<script type="text/javascript">
    // for IE
    $pnp.setup({
        headers: {
            "Accept": "application/json;odata=verbose"
        }
    });

    $pnp.sp.search('contenttype:newspage').then(function(result) {
        alert(result);
        })
        .catch(function(error){ 
            document.getElementById("searchDiv").innerHTML = error;
        });

    // current user
    $pnp.sp.profiles.myProperties.get().then(function(result) {
        var propValue = "";
        propValue += result.DisplayName + "<br/>";
        propValue += result.Email + "<br/>";
        document.getElementById("user").innerHTML = propValue;
    })
    .catch(function(error){ 
            document.getElementById("user").innerHTML = error;
    });
</script>

**Comment by joukamp** [comment]
Heres some more from ther error:

"ReferenceError: 'Symbol' is undefined\n   
at value (https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.min.js:1:19622)\n   
at e (https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.min.js:1:19115)\n   
at Anonymous function (https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.min.js:1:18985)\n   
at tryCatch (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:414:5)\n   
at invokeCallback (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:429:5)\n   
at publish (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:397:7)\n   
at flush (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:125:5)"

**Comment by patrick-rodgers** [comment]
Can you run it using just pnp.js - that will let us get an actual line number. This may have something to do with an iterator, but unsure why yet. Or maybe something else entirely. I have successfully used the library in IE 10 & 11 for testing and it works for me, so I am having some trouble pinpointing what might be happening.

**Comment by joukamp** [comment]
using pnp.js:

"ReferenceError: 'Symbol' is undefined\n   
at formatSearchResults (https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.js:1674:23)\n   
at SearchResults (https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.js:1650:10)\n   
at Anonymous function (https://cdnjs.cloudflare.com/ajax/libs/sp-pnp-js/2.0.1/pnp.js:1628:18)\n   
at tryCatch (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:414:5)\n   
at invokeCallback (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:429:5)\n   
at publish (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:397:7)\n   
at flush (https://cdnjs.cloudflare.com/ajax/libs/es6-promise/4.0.5/es6-promise.auto.js:125:5)"

**Comment by patrick-rodgers** [comment]
Thanks! So something in the pipeline is adding that block of code (it isn't literally in our source). But that line is:

```JavaScript
for (var _iterator = tempResults[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = true) {
```

That is trying to access Symbol. I'll have a look and see why might be the issue. Just to confirm could you give me the exact IE 11 version (shouldn't matter but maybe). Mean while I can start looking into this. Thanks again for reporting it and helping to diagnose.

**Comment by joukamp** [comment]
IE version: 
11.576.14393.0
Update versions: 11.0.38 (KB3203621)
Product ID: 00150-20000-00003-AA459

**Comment by joukamp** [comment]
Hi,

If I add:

<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/6.23.0/polyfill.js"></script>
at top of my script it works!
I don't know why, I just tried all things.

**Comment by patrick-rodgers** [comment]
Thanks @joukamp - I didn't figure out exactly what was wrong. However I checked and Symbol does not appear in the new dist files that will be released on Friday (3/10/2017) so this will be fixed in the 2.0.2 release, likely as part of the rollback to es5. So going to close this as you have a workaround for now and as fixed in upcoming release. Appreciate you working with me on troubleshooting.
